File: easy/ASCII Art/Perl/ASCII Art.pl
use strict;
use warnings;
use 5.32.1;

select(STDOUT); $| = 1; # DO NOT REMOVE

chomp(my $l = <STDIN>);
chomp(my $h = <STDIN>);
chomp(my $t = <STDIN>);
$t = uc $t;

my @asciiArt;
for my $i (0..$h-1) {
    chomp(my $row = <STDIN>);
    push @asciiArt, $row;
}

my @result = ('') x $h;

# Precomputing
my @indices = map { ($_ ge 'A' && $_ le 'Z') ? ord($_) - ord('A') : 26 } split //, $t;

foreach my $index (@indices) {
    for my $i (0..$h-1) {
        my $start = $index * $l;
        my $end = $start + $l;
        $result[$i] .= substr($asciiArt[$i], $start, $l);
    }
}

for my $i (0..$h-1) {
    print "$result[$i]\n";
}

File: easy/ASCII Art/PHP/ASCII Art.php
<?php
fscanf(STDIN, "%d", $l);
fscanf(STDIN, "%d", $h);
$t = strtoupper(stream_get_line(STDIN, 256 + 1, "\n"));

$asciiArt = [];
for ($i = 0; $i < $h; $i++) {
    $asciiArt[$i] = stream_get_line(STDIN, 1024 + 1, "\n");
}

$result = array_fill(0, $h, '');

$chars = str_split($t);
foreach ($chars as $c) {
    $index = ctype_alpha($c) ? ord($c) - ord('A') : 26;

    for ($i = 0; $i < $h; $i++) {
        $start = $index * $l;
        $result[$i] .= substr($asciiArt[$i], $start, $l);
    }
}

foreach ($result as $line) {
    echo $line . "\n";
}
?>
